### DISCUSSION TITLE
Table with warning for the whole row width

### DISCUSSION THREAD

**Discussion by myhlibovych** [twilio-labs/paste#1296] [Q&A] [answered]
Hello!

I'm building the new functionality for Regulatory Bundles, and I've come up with a solution where I need to display warning inside the table row with action items. I haven't found such pattern in Paste design system. 

I chose this approach because this shows connection between current regulatory bundle and new one (all numbers will be assigned to new one after it gets approved, it clearly shows those rows that require the most attention, and it displays additional data that is specific to the draft bundle).

Does it fall in the guidelines or should I look for another option?

![bundle list - fork created](https://user-images.githubusercontent.com/71466379/111617119-636bb700-87eb-11eb-9028-781254a507cf.png)

**Comment by myhlibovych** [comment]
I've also removed bottom border from the warning because I wanted to avoid extra visual weight because of the usage of the warning in the table. Can I keep it this way or should I use the standard one with the bottom border?

**Comment by TheSisb** [comment, marked as answer]
Hi @myhlibovych,

> I chose this approach because this shows connection between current regulatory bundle and new one (all numbers will be assigned to new one after it gets approved, it clearly shows those rows that require the most attention, and it displays additional data that is specific to the draft bundle).  Does it fall in the guidelines or should I look for another option?

For Tables, we advise against injecting rows like that because it is a poor user experience with screenreader software. 
I understand the desire to show a connection between the current row and the warning. You may want to consider an alternate treatment, such as: https://www.lightningdesignsystem.com/components/data-tables/#Edited-cell-with-focused-row-level-error-(unsaved)
 
> I've also removed bottom border from the warning because I wanted to avoid extra visual weight because of the usage of the warning in the table. Can I keep it this way or should I use the standard one with the bottom border?

For our styled Components (unlike our un-styled, Primitive Components), making changes to the design is very expensive. It means that you're ejecting from the paved path and will have problems getting design updates in the future in any reasonable way. We strongly recommend leaving the design intact.  If there's an issue with the existing Component's design it is best to open up a new discussion thread to discuss how we can improve it generically.

Hope this helps!

**Reply by corinne-nagel** [reply]
Hi @myhlibovych , following up here, did you get the info you needed to make a decision?

**Reply by gloriliale** [reply]
Hi @myhlibovych! We hope this info helped in figuring out the best way forward- let us know if there's anything else you need.
